sympy/sets/fancysets.py

```diff
@@ -472,14 +478,17 @@ def __new__(cls, *args):
         if start.is_infinite:
             end = stop
         else:
-            ref = start if start.is_finite else stop
-            n = ceiling((stop - ref)/step)
-            if n <= 0:
-                # null Range
-                start = end = S.Zero
-                step = S.One
+            if not all(w.is_number for w in [start, stop, step]):
+                end = stop
             else:
-                end = ref + n*step
+                ref = start if start.is_finite else stop
+                n = ceiling((stop - ref)/step)
```

Comment:
I think the check for Float can delay until here where we test:
```
if step.atoms(Float) or (stop - ref).atoms(Float):
    raise error
```
This allows `Range(Float(1), Float(1)+3/S(10),1/S(10)) -> [1.0, 1.1, 1.2]`

Reply:
We can do that, however, the case is, we have to check that if there is a symbol present in start, stop, step then we should not raise the error. For example, if `(n + 0.5).atoms(Float)` is done, then it will raise according to the logic you gave above, however, `n + 0.5`, can be made an integer, by substituting n with 0.5, 1.5, 2.5, .. etc.

Reply:
Adding two Floats will not create an Integer (nor does `Float(1) == 1`) so I think we're ok. Ah..but substituting `n:-.5` will cancel out the Float so yes, no error if there are `free_symbols` in the entities:
`if any(i.is_number and i.has(Float) for i in (step, stop - ref)): raise error`?